flowmachine/flowmachine/features/subscriber/iterative_median_filter.py

```diff
@@ -0,0 +1,92 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+from typing import List
+
+from flowmachine.core import Query
+
+
+class IterativeMedianFilter(Query):
+    """
+    Applies an iterative median filter
+
+    Parameters
+    ----------
+    query_to_filter : Query
+        Query to apply iterated median filter to.
+    filter_window_size : int
+        Size of filter window - must be odd, positive and more than 1.
+    column_to_filter : str, default 'value'
+        Column to apply the filter to.
+    partition_column : str, default 'subscriber'
+        Column to use for partitioning. May be none, in which case no partitioning is applied.
+    order_column : str, default datetime
+        Column to use for ordering within partitions. May be none, in which case no ordering is applied.
+    """
+
+    def __init__(
+        self,
+        *,
+        query_to_filter: Query,
+        filter_window_size: int,
+        column_to_filter: str = "value",
+        partition_column: str = "subscriber",
+        order_column: str = "datetime",
+    ):
+
+        self.query_to_filter = query_to_filter
+        self.filter_window_size = filter_window_size
+        if not (filter_window_size > 1):
+            raise ValueError("filter_window_size must be positive and more than 1.")
+        if (filter_window_size % 2) == 0:
+            raise ValueError("filter_window_size must be odd.")
+        self.column_to_filter = column_to_filter
+        if column_to_filter not in query_to_filter.column_names:
```

Comment:
Could be useful (in a separate PR) to create a helper method for this check to avoid some repetition here?

E.g.:
```
query_to_filter.verify_column_exists(column_to_filter)
query_to_filter.verify_column_exists(partition_column)
query_to_filter.verify_column_exists(order_column)
```